src/core/reference/include/ngraph/runtime/reference/irdft.hpp

```diff
@@ -0,0 +1,45 @@
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <functional>
+#include <map>
+#include <ngraph/runtime/host_tensor.hpp>
+#include <vector>
+
+#include "ngraph/node.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/ops.hpp"
+#include "ngraph/shape_util.hpp"
+
+namespace ngraph {
+namespace runtime {
+namespace reference {
+void irdft(const std::vector<float>& input_data,
+           const Shape& input_data_shape,
+           const std::vector<int64_t>& axes_data,
+           float* irdft_result,
+           const Shape& fft_output_shape,
+           int64_t last_signal_size);
```

Comment:
It misses info what should be the size of the buffer `irdft_result` - how to calculate it.
I don't think it's a safe approach anyway ..

Reply:
good point, changed to pass `irdft_output_shape` as argument and start using `shape_size(irdft_output_shape)` during irdft_result manipulation